tc/sgx/trusted_worker_manager/enclave/signup_enclave_wpe.cpp

```diff
@@ -44,6 +44,17 @@ static void CreateSignupReportDataWPE(const uint8_t* ext_data,
     EnclaveData* enclave_data,
     sgx_report_data_t* report_data);
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+tcf_err_t ecall_GenerateNonce(uint8_t* out_nonce, size_t in_nonce_size) {
+    tcf_err_t result = TCF_SUCCESS;
+    tcf::error::ThrowIf<tcf::error::ValueError>(in_nonce_size <= 0,
+        "Nonce size should be positive value");
+
+    sgx_status_t ret = sgx_read_rand(out_nonce, in_nonce_size);
```

Comment:
Can we use openSSL function to generate nonce. That way this function can be SGX independent.

Reply:
Ok. I used crypto utility function which uses openssl/sgxssl to generate random nonce.